### PR TITLE
Armada-661 Migrate the jobbergate api and jobbergate cli packages to Pypi

### PR DESCRIPTION
**Pull request by fschuch**
#### What
Migrate the jobbergate api and jobbergate cli packages to Pypi, including:

* GitHub actions to publish them on new tags;
* Release scripts to help in the process.

#### Why
Since we are publishing the source code publicly, there's no reason we shouldn't also publish package releases to PyPI (as we do with Armasec).

`Task`: https://app.clickup.com/t/18022949/ARMADA-661

---

#### Peer Review
Please follow the upstream omnivector documentation concerning
[peer-review guidelines](https://github.com/omnivector-solutions/Documentation/blob/main/Contributing/pr_review_standards.md#peer-review).

### CODE REVIEW

**Comment by dusktreader** [comment]
Task linked: [ARMADA-661 Migrate the jobbergate-api and jobbergate-cli packages](https://app.clickup.com/t/18022949/ARMADA-661)

**Comment by fschuch** [comment]
The workflow for a new release is as follows:

* The [bump_version](https://github.com/omnivector-solutions/jobbergate/blob/5541ab9d2ca12bf468237ea35c4ce3cc84444202/.github/workflows/bump_version.yaml) action accepts as an argument anything that works with `poetry version` (like `patch`, `minor`, `major`, `prepatch`, `preminor`, `premajor`, and `prerelease`). It can be used on demand to apply the following procedures to all subprojects (API and CLI for now):

    * Use `poetry version <arg>` to bump the sub-project's version;
    * Create a new header at the changelog for the new version;
    * Check if the new versions are the same for all sub-projects;
    * Create a branch named `release/<new-version>` and a draft PR including the changes above. The PR can be marked as ready to review and in fact be reviewed by the team.
 
   Note: The action can be triggered from any branch, which is nice to create prereleases, but it is a good practice to make releases only from the main branch (that is the default behavior, but there is no checks for it).
 
 * Once the PR is reviewed, approved, and merged, [tag_on_merged_pull_request](https://github.com/omnivector-solutions/jobbergate/blob/5541ab9d2ca12bf468237ea35c4ce3cc84444202/.github/workflows/tag_on_merged_pull_request.yaml) is automatically triggered to create and push a new tag named `<new-version>`;
 
 * The new tag triggers [publish_on_tag](https://github.com/omnivector-solutions/jobbergate/blob/5541ab9d2ca12bf468237ea35c4ce3cc84444202/.github/workflows/publish_on_tag.yaml), that will:
 
   * Double-check if the version for each sub-project matches the tag;
   * Deploy all sub-projects to PyPI.

**Comment by fschuch** [comment]
The workflow for a new release is as follows:

* The [bump_version](https://github.com/omnivector-solutions/jobbergate/blob/5541ab9d2ca12bf468237ea35c4ce3cc84444202/.github/workflows/bump_version.yaml) action accepts as an argument anything that works with `poetry version` (like `patch`, `minor`, `major`, `prepatch`, `preminor`, `premajor`, and `prerelease`). It can be used on demand to apply the following procedures to all subprojects (API and CLI for now):

    * Use `poetry version <arg>` to bump the sub-project's version;
    * Create a new header at the changelog for the new version;
    * Check if the new versions are the same for all sub-projects;
    * Create a branch named `release/<new-version>` and a draft PR including the changes above. The PR can be marked as ready to review and in fact be reviewed by the team.
 
   Note: The action can be triggered from any branch, which is nice to create prereleases, but it is a good practice to make releases only from the main branch (that is the default behavior, but there is no checks for it).
 
 * Once the PR is reviewed, approved, and merged, [tag_on_merged_pull_request](https://github.com/omnivector-solutions/jobbergate/blob/5541ab9d2ca12bf468237ea35c4ce3cc84444202/.github/workflows/tag_on_merged_pull_request.yaml) is automatically triggered to create and push a new tag named `<new-version>`;
 
 * The new tag triggers [publish_on_tag](https://github.com/omnivector-solutions/jobbergate/blob/5541ab9d2ca12bf468237ea35c4ce3cc84444202/.github/workflows/publish_on_tag.yaml), that will:
 
   * Double-check if the version for each sub-project matches the tag;
   * Deploy all sub-projects to PyPI.

File: .github/workflows/bump_version.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
+name: Bump Jobbergate Version
```
Comment (fschuch) [inline .github/workflows/bump_version.yaml:1]: Should we rename it to `Prepare for Release`? It would be more straightforward for the devs

Comment (dusktreader) [reply .github/workflows/bump_version.yaml:1]: Yes, please.

Comment (fschuch) [reply .github/workflows/bump_version.yaml]: solved in [1077fbf](https://github.com/omnivector-solutions/jobbergate/pull/124/commits/1077fbfb52cada73757b00a695aca0fd98830741)

**Review by dusktreader — APPROVED** [review]

**Pull request merged**